litellm/llms/notdiamond.py

```diff
@@ -0,0 +1,253 @@
+    "anthropic/claude-3-opus-20240229": "claude-3-opus-20240229",
+    "anthropic/claude-3-sonnet-20240229": "claude-3-sonnet-20240229",
+    "anthropic/claude-3-5-sonnet-20240620": "claude-3-5-sonnet-20240620",
+    "anthropic/claude-3-haiku-20240307": "claude-3-haiku-20240307",
+    # mistral
+    "mistral/mistral-large-latest": "mistral/mistral-large-latest",
+    "mistral/mistral-medium-latest": "mistral/mistral-medium-latest",
+    "mistral/mistral-small-latest": "mistral/mistral-small-latest",
+    "mistral/codestral-latest": "mistral/codestral-latest",
+    "mistral/open-mistral-7b": "mistral/open-mistral-7b",
+    "mistral/open-mixtral-8x7b": "mistral/open-mixtral-8x7b",
+    "mistral/open-mixtral-8x22b": "mistral/open-mixtral-8x22b",
+    # perplexity
+    "perplexity/llama-3-sonar-large-32k-online": "perplexity/llama-3-sonar-large-32k-online",
+    # cohere
+    "cohere/command-r": "cohere_chat/command-r",
+    "cohere/command-r-plus": "cohere_chat/command-r-plus",
+    # google
+    "google/gemini-pro": "gemini/gemini-pro",
+    "google/gemini-1.5-pro-latest": "gemini/gemini-1.5-pro-latest",
+    "google/gemini-1.5-flash-latest": "gemini/gemini-1.5-flash-latest",
+    "google/gemini-1.0-pro-latest": "gemini/gemini-pro",
+    # replicate
+    "replicate/mistral-7b-instruct-v0.2": "replicate/mistralai/mistral-7b-instruct-v0.2",
+    "replicate/mixtral-8x7b-instruct-v0.1": "replicate/mistralai/mixtral-8x7b-instruct-v0.1",
+    "replicate/meta-llama-3-70b-instruct": "replicate/meta/meta-llama-3-70b-instruct",
+    "replicate/meta-llama-3-8b-instruct": "replicate/meta/meta-llama-3-8b-instruct",
+    # togetherai
+    "togetherai/Mistral-7B-Instruct-v0.2": "together_ai/mistralai/Mistral-7B-Instruct-v0.2",
+    "togetherai/Mixtral-8x7B-Instruct-v0.1": "together_ai/mistralai/Mixtral-8x7B-Instruct-v0.1",
+    "togetherai/Mixtral-8x22B-Instruct-v0.1": "together_ai/mistralai/Mixtral-8x22B-Instruct-v0.1",
+    "togetherai/Phind-CodeLlama-34B-v2": "together_ai/Phind/Phind-CodeLlama-34B-v2",
+    "togetherai/Llama-3-70b-chat-hf": "together_ai/meta-llama/Llama-3-70b-chat-hf",
+    "togetherai/Llama-3-8b-chat-hf": "together_ai/meta-llama/Llama-3-8b-chat-hf",
+    "togetherai/Qwen2-72B-Instruct": "together_ai/Qwen/Qwen2-72B-Instruct",
+}
+
+
+class NotDiamondError(Exception):
+    def __init__(self, status_code, message):
+        self.status_code = status_code
+        self.message = message
+        self.request = httpx.Request(
+            method="POST", url="https://not-diamond-server.onrender.com/v2/optimizer/router"
+        )
+        self.response = httpx.Response(status_code=status_code, request=self.request)
+        super().__init__(
+            self.message
+        )  # Call the base class constructor with the parameters it needs
+
+
+class NotDiamondConfig:
+    # llm_providers requires at least one provider, setting to gpt-3.5-turbo as default
+    llm_providers: Optional[List[Dict[str, str]]] = [
```

Comment:
You don't need to set this default value both here and in `__init__(llm_providers)`. I'd suggest choosing one or the other. 

I'm also not sure about the choice of default value here. Imagine that the user doesn't pass in a `llm_providers`. Should we either (1) raise an exception or (2) default to all providers supported by Not Diamond? What are your thoughts?

Reply:
Initial thought was to default to all providers supported by Not Diamond. But that could lead to errors, such as missing API keys. I am now more inclined to just raise an exception if we don't want to impose a default model. @r0ymanesco would love to hear your thoughts on this too!

Reply:
Agree we shouldn’t allow no providers to be specified. It’s the same thing with the API